pkg/config/user_config.go

```diff
@@ -195,22 +195,22 @@ type GitConfig struct {
 	// If true, do not allow force pushes
 	DisableForcePushing bool `yaml:"disableForcePushing"`
 	// See https://github.com/jesseduffield/lazygit/blob/master/docs/Config.md#predefined-commit-message-prefix
-	CommitPrefixes map[string]CommitPrefixConfig `yaml:"commitPrefixes"`
+	CommitPrefixes map[string]CommitPrefixConfig `yaml:"commitPrefixes" jsonschema:"additionalProperties=false"`
 	// If true, parse emoji strings in commit messages e.g. render :rocket: as 🚀
 	// (This should really be under 'gui', not 'git')
 	ParseEmoji bool `yaml:"parseEmoji"`
 	// Config for showing the log in the commits view
-	Log LogConfig `yaml:"log"`
+	Log LogConfig `yaml:"log" jsonschema:"additionalProperties=false"`
 }
 
 type PagingConfig struct {
 	// Value of the --color arg in the git diff command. Some pagers want this to be set to 'always' and some want it set to 'never'
-	ColorArg string `yaml:"colorArg"`
+	ColorArg string `yaml:"colorArg" jsonschema:"enum=always,enum=never"`
 	// e.g.
 	// diff-so-fancy
 	// delta --dark --paging=never
 	// ydiff -p cat -s --wrap --width={{columnWidth}}
-	Pager string `yaml:"pager"`
+	Pager string `yaml:"pager" jsonschema:"example=delta --dark --paging=never,example=diff-so-fancy,example=ydiff -p cat -s --wrap --width={{columnWidth}},minLength=1"`
```

Comment:
These examples don't seem to work; the only one that shows up in the schema is "diff-so-fancy". I'm not sure this is because of the whitespace or because of the `=` in the values. I also couldn't figure out how to fix it; adding single quotes around the values doesn't help.

Reply:
could be fixed using this approach: cf48f4d94d1104785d1457773773243719065e92

otherwise we can just remove this

Reply:
Nice, I like this.

Reply:
Do you think we should just keep these `jsonschema` relevant functions in the `user_config` file or should we put them in a different file in the same package? 

I guess we could also define the type and the extend function in the `jsonschema` package and import it into `user_config`

Reply:
Since we define most of the rest of what goes into the schema in user_config.go, I think these extra things might as well be there, too. Not a strong opinion, I could go either way, and since @jesseduffield usually has a better sense of what should go where, I'll let him decide.